tests_and_analysis/test/euphonic_test/test_force_constants_calculate_qpt_phonons_castep.py

```diff
@@ -0,0 +1,492 @@
+                 1.8361944616e-04, 1.8361944616e-04, 4.3310694904e-04,
+                 4.0094292783e-03, 4.0298656188e-03, 7.0857269505e-03,
+                 7.0857269505e-03, 7.1044827434e-03, 7.1044827434e-03],
+                [1.8594596548e-05, 2.8377223801e-05, 4.9463914519e-05,
+                 1.8632830014e-04, 1.9060940094e-04, 4.3314392073e-04,
+                 4.0093920384e-03, 4.0298240625e-03, 7.0856315682e-03,
+                 7.0859645027e-03, 7.1043776174e-03, 7.1046906727e-03],
+                [2.1681107372e-03, 2.2111728203e-03, 2.8193681109e-03,
+                 2.8197547098e-03, 2.9049711018e-03, 2.9141516000e-03,
+                 6.0276999661e-03, 6.0316367945e-03, 6.0977872708e-03,
+                 6.1104623404e-03, 6.3230407709e-03, 6.3262959354e-03],
+                [7.6869781440e-04, 8.8171407111e-04, 2.1005260482e-03,
+                 2.1043510112e-03, 3.5723508344e-03, 3.5925812023e-03,
+                 3.7679565852e-03, 3.7804526859e-03, 6.6594311404e-03,
+                 6.6651464171e-03, 7.2007012512e-03, 7.2132353897e-03],
+                [1.4065426444e-04, 1.4065426444e-04, 1.4065426444e-04,
+                 1.4065426444e-04, 3.2305904460e-04, 3.2305904460e-04,
+                 4.0222682768e-03, 4.0222682768e-03, 7.1591503492e-03,
+                 7.1591503492e-03, 7.1591503492e-03, 7.1591503492e-03]
+            ]) * ureg('hartree')
+        }
+    }
+
+    materials = [
+        (
+            "LZO", 'La2Zr2O7.castep_bin',
+            np.array([  # Qpoints
+                [-1.00, 9.35, 3.35],
+                [-1.00, 9.00, 3.00]
+            ])
+        ),
+        (
+            "graphite", "graphite.castep_bin",
+            np.array([  # Qpoints
+                [0.00, 0.00, 0.00],
+                [0.001949, 0.001949, 0.00],
+                [0.50, 0.00, 0.00],
+                [0.25, 0.00, 0.00],
+                [0.00, 0.00, 0.50]
+            ])
+        )
+    ]
+
+    kwargs = [
+        {"use_c": False, "fall_back_on_python": True, "n_threads": 1},
+        {"use_c": True, "fall_back_on_python": False, "n_threads": 1},
+        {"use_c": True, "fall_back_on_python": False, "n_threads": 2},
+        {
+            "use_c": False, "fall_back_on_python": True,
+            "n_threads": 1, "asr": "realspace"
+        },
+        {
+            "use_c": True, "fall_back_on_python": False,
+            "n_threads": 1, "asr": "realspace"
+        },
+        {
+            "use_c": True, "fall_back_on_python": False,
+            "n_threads": 2, "asr": "realspace"
+        }
+    ]
+
+    @pytest.fixture(
+        params=list(itertools.product(materials, kwargs))
+    )
+    def calculate_qpoint_phonon_modes(self, request):
+        material, kwargs = request.param
+        material_name, castep_bin_file, qpts = material
+        filename = os.path.join(self.path, material_name, castep_bin_file)
+        return ForceConstants.from_castep(filename), qpts, material_name, kwargs
```

Comment:
For testing functions we should be reading directly from `.json` files rather than using `from_castep/phonopy`

Reply:
Refactored to do this